capstone-rs/src/capstone.rs

```diff
@@ -423,3 +435,64 @@ impl Drop for Capstone {
         unsafe { cs_close(&mut self.csh()) };
     }
 }
+
+/// Structure to handle iterative disassembly
+///
+/// Create with a capstone instance `get_disasm_iter()`
+///
+pub struct DisasmIter<'a> {
+    insn: *mut cs_insn, // space for current instruction to be processed
+    csh: *mut c_void,   // reference to the the capstone handle required by disasm_iter
+    offset: usize,
+    addr: u64,
+    _covariant: PhantomData<&'a ()>, // used to make sure DIasmIter lifetime doesn't exceed Capstone's lifetime
+}
+
+impl<'a> Drop for DisasmIter<'a> {
+    fn drop(&mut self) {
+        unsafe { cs_free(self.insn, 1) };
+    }
+}
+
+impl<'a> DisasmIter<'a> {
+    /// Used to continue to the next instruction without jumping
+    ///
+    /// usage shown in examples/recursive.rs
+    pub fn disasm_iter_continue(&mut self, code: &[u8]) -> CsResult<Insn> {
+        self.disasm_iter(code, self.offset, self.addr)
+    }
+
+    /// Used to start the iterative disassembly
+    ///
+    /// usage shown in examples/recursive.rs
+    pub fn disasm_iter(&mut self, code: &[u8], offset: usize, addr: u64) -> CsResult<Insn> {
+        let code_len = code.len();
+        let code_ptr = &mut code[offset..].as_ptr();
+        let mut c = code_len - offset;
```

Comment:
Avoid one-word variable names. Perhaps something like `new_code_ptr` and `new_addr`?